tests/metrics/functional/test_classification.py

```diff
@@ -35,28 +38,43 @@
 )
 
 
-@pytest.mark.parametrize(['sklearn_metric', 'torch_metric'], [
-    pytest.param(sk_accuracy, accuracy, id='accuracy'),
-    pytest.param(partial(sk_precision, average='macro'), precision, id='precision'),
-    pytest.param(partial(sk_recall, average='macro'), recall, id='recall'),
-    pytest.param(partial(sk_f1_score, average='macro'), f1_score, id='f1_score'),
-    pytest.param(partial(sk_fbeta_score, average='macro', beta=2), partial(fbeta_score, beta=2), id='fbeta_score'),
-    pytest.param(sk_confusion_matrix, confusion_matrix, id='confusion_matrix')
+@pytest.mark.parametrize(['sklearn_metric', 'torch_metric', 'only_binary'], [
+    pytest.param(sk_accuracy, accuracy, False, id='accuracy'),
+    pytest.param(partial(sk_precision, average='macro'), precision, False, id='precision'),
+    pytest.param(partial(sk_recall, average='macro'), recall, False, id='recall'),
+    pytest.param(partial(sk_f1_score, average='macro'), f1_score, False, id='f1_score'),
+    pytest.param(partial(sk_fbeta_score, average='macro', beta=2),
+                 partial(fbeta_score, beta=2), False, id='fbeta_score'),
+    pytest.param(sk_confusion_matrix, confusion_matrix, False, id='confusion_matrix'),
+    pytest.param(sk_roc_curve, roc, True, id='roc'),
+    pytest.param(sk_precision_recall_curve, precision_recall_curve, True, id='precision_recall_curve'),
+    pytest.param(sk_roc_auc_score, auroc, True, id='auroc')
 ])
-def test_against_sklearn(sklearn_metric, torch_metric):
+def test_against_sklearn(sklearn_metric, torch_metric, only_binary):
```

Comment:
```suggestion
def test_against_sklearn(sklearn_metric, torch_metric, only_binary=False):
```
maybe this works (not sure), then you don't have to add it to all metrics above, only the few ones that are binary only

Reply:
the test has grown more complex now with the additional if statements. We have to be careful not to overload the tests with too much logic, or we have to test the test itself... if you know what I mean :)

Reply:
not a major concern, just something to look out in the future. I don't have a good suggestion as how to avoid it in this specific case.